webrtc.html

```diff
@@ -3932,52 +3901,178 @@ <h4 id="sec.identity-proxy-communications">Identity Provider
       <section>
         <h4 id="sec.create-identity-proxy">Instantiating an IdP Proxy</h4>
 
-        <p>The browser creates an IdP proxy by loading an isolated, invisible
-        IFRAME with HTML content from the IdP URI. The URI for the IdP is a
+        <p>In order to communicate with the IdP, the user agent loads the IdP
+        JavaScript from the IdP.   The URI for the IdP script is a
         well-known URI formed from the <q>domain</q> and <q>protocol</q>
         fields, as specified in [[!RTCWEB-SECURITY-ARCH]].</p>
 
-        <p>When an IdP proxy is requiured, the browser performs the following
-        steps:</p>
+        <p>The IdP MAY generate an HTTP redirect to another "https" origin, the
+        browser MUST treat a redirect to any other scheme as a fatal error.</p>
+
+        <p>The user agent instantiates an isolated interpreted context, a
+        JavaScript <a
+        href="http://people.mozilla.org/~jorendorff/es6-draft.html#sec-code-realms"><dfn>realm</dfn></a>
+        that operates in the origin of the loaded JavaScript. Note that a
+        redirect will change the origin of the loaded script.</p>
+
+        <p>The <a>realm</a> is populated with a global that implements
+        <code>WorkerGlobalScope</code> [[!WEBWORKERS]].</p>
+
+        <p>The user agent provides instance of
+        <code><a>RTCIdentityProviderRegistrar</a></code> named
+        <var>rtcIdentityProvider</var> in the global scope of the <a>realm</a>.
+        This object is used by the IdP to interact with the user agent.</p>
+
+        <p>Since it is an essential security property of the web sandbox that a
+        page is unable to insert objects into content from another origin, using
+        the global <var>rtcIdentityProvider</var> instance ensures that the IdP
```

Comment:
This doesn't seem maximally clear. Can you rewrite it to make it clearer what the property is?